### ISSUE TITLE
System.Threading.Tests.Perf_Timer.ScheduleManyThenDisposeMany has regressed on 64 bit Windows

### ISSUE THREAD

**Issue by adamsitnik** [dotnet/runtime#67972]
`System.Threading.Tests.Perf_Timer.ScheduleManyThenDisposeMany` has regressed on 64 bit Windows (both x64 and arm64).

[Reporting System](https://pvscmdupload.blob.core.windows.net/reports/allTestHistory%2frefs%2fheads%2fmain_x64_Windows%2010.0.18362%2fSystem.Threading.Tests.Perf_Timer.ScheduleManyThenDisposeMany.html)

![image](https://user-images.githubusercontent.com/6011991/163239492-dc1c0bbb-d7e4-40b0-9594-f4b37b1bff58.png)

Looking at the [diff](https://github.com/dotnet/runtime/compare/a4a6d30517b5b141b3934814100dfb5f20f07d5d...801e81ee8a5d934a50d91d5dca9ad62a4c708bff) which contains only 2 commits I am guessing that it was caused by #60248 (cc @PeterSolMS) but I am not sure.

Repro:

```cmd
git clone https://github.com/dotnet/performance.git
py .\performance\scripts\benchmarks_monthly.py net7.0-preview2 net7.0-preview3 --filter System.Threading.Tests.Perf_Timer.ScheduleManyThenDisposeMany
```

<details>

## System.Threading.Tests.Perf_Timer.ScheduleManyThenDisposeMany

| Result | Ratio | Operating System      | Bit   | Processor Name                                  | Modality|
| ------ | -----:| --------------------- | ----- | ----------------------------------------------- | -------- |
| Same   |  1.06 | alpine 3.12           | Arm64 | Unknown processor                               |         |
| Same   |  1.05 | ubuntu 18.04          | Arm64 | Unknown processor                               |         |
| Slower |  0.86 | Windows 10            | Arm64 | Microsoft SQ1 3.0 GHz                           | bimodal |
| Slower |  0.87 | Windows 11            | Arm64 | Microsoft SQ1 3.0 GHz                           | several?|
| Slower |  0.76 | macOS Monterey 12.3   | Arm64 | Apple M1 Max                                    |         |
| Slower |  0.81 | Windows 11            | X64   | AMD Ryzen Threadripper PRO 3945WX 12-Cores      |         |
| Slower |  0.80 | Windows 11            | X64   | Intel Core i7-8700 CPU 3.20GHz (Coffee Lake)    |         |
| Slower |  0.82 | Windows 11            | X64   | Intel Core i9-9900T CPU 2.10GHz                 |         |
| Same   |  0.97 | alpine 3.13           | X64   | Intel Core i7-7700 CPU 3.60GHz (Kaby Lake)      |         |
| Same   |  1.09 | centos 7              | X64   | Intel Xeon CPU E5530 2.40GHz                    |         |
| Same   |  0.97 | debian 11             | X64   | Intel Core i7-7700 CPU 3.60GHz (Kaby Lake)      | bimodal |
| Same   |  0.99 | opensuse-leap 15.3    | X64   | Intel Core i7-7700 CPU 3.60GHz (Kaby Lake)      |         |
| Same   |  1.10 | ubuntu 18.04          | X64   | Intel Xeon CPU E5-1650 v4 3.60GHz               | bimodal |
| Faster |  1.21 | ubuntu 18.04          | X64   | Intel Core i7-2720QM CPU 2.20GHz (Sandy Bridge) |         |
| Same   |  1.08 | Windows 10            | Arm   | Microsoft SQ1 3.0 GHz                           |         |
| Same   |  1.02 | ubuntu 18.04          | Arm   | ARMv7 Processor rev 3 (v7l)                     | several?|
| Same   |  0.92 | Windows 10            | X86   | Intel Xeon CPU E5-1650 v4 3.60GHz               | several?|
| Same   |  0.95 | Windows 11            | X86   | AMD Ryzen Threadripper PRO 3945WX 12-Cores      | bimodal |
| Faster |  1.70 | Windows 11            | X86   | Intel Core i7-10510U CPU 1.80GHz                |         |
| Same   |  1.03 | Windows 7 SP1         | X86   | Intel Core i7-7700 CPU 3.60GHz (Kaby Lake)      | bimodal |
| Slower |  0.88 | macOS Monterey 12.2.1 | X64   | Intel Core i7-5557U CPU 3.10GHz (Broadwell)     |         |
| Slower |  0.88 | macOS Monterey 12.3.1 | X64   | Intel Core i7-4870HQ CPU 2.50GHz (Haswell)      | bimodal |

</details>

**Comment by PeterSolMS** [comment]
I still see a regression in the numbers reported - for .NET 6.0 I see 433.5 msec / iteration while for 7.0 I see 495.9.

However, profiles show that the GC pauses are actually smaller for both 7.0 with segments and 7.0 with regions, so the root cause of the regression is still a bit of a mystery.

The gen 1 accounting fix does play a role though - in 6.0, we have alternating gen 0 (190 of them) and gen 1 (also 190), where gen 0 mean pause time is 5.4 msecs and gen 1 mean pause time is 23.8 msecs.

In .NET 7.0 we see mainly gen 1 GC (380 of them) with a mean pause time of 10 msecs, and 20 gen 0 GC with a mean pause time of 8.6 msecs.

Overall pause time in .NET 6.0 is 14.2311 seconds, while for .NET 7.0 I get 12.1019 seconds. This test does a lot of induced GCs -  in fact they dominate the pause time. Subtracting the induced GC pause time from the total pause time gives 6.271 seconds for .NET 6.0 and 4.704 seconds for .NET 7.0.

The proposed fix to more gradually decommit ephemeral space has a small effect - time per iteration goes to 482.8 msecs with this fix. Decommitting nothing at all is even better - 457.7 msecs, so decommit policy certainly matters, and is likely something we should work on in .NET 8.0.

The test shows unusual behavior in that for most of the gen 0 and gen 1 collections, survival rate is 100%. This causes the time in GC to be about 70%.

**Comment by mangod9** [comment]
Based on Peter's analysis this doesnt seem to be of much concern.
